app_test.go

```diff
@@ -84,6 +84,16 @@ func TestNewApp(t *testing.T) {
 	})
 }
 
+var errCount = 0
+
+type testErrHandler struct {
+	count int
+}
+
+func (eh testErrHandler) HandleError(err error) {
+	errCount++
```

Comment:
I believe that you intended to have `eh.count++` and remove the global `errCount` var.
Then your test below can modify the assertion to be `assert.Equal(t, 1, errHandler.count)`

Reply:
I changed both the counter and the strings to fields in the handlers 😬